### ISSUE TITLE
PossiblyNullArrayAccess inside "not null" condition

### ISSUE THREAD

**Issue by bkdotcom** [vimeo/psalm#3351]
https://psalm.dev/r/c2f40fbabd

```php
$error = \error_get_last();
$isFatalError = $error !== null && $error['type'] & (E_ERROR | E_PARSE | E_COMPILE_ERROR | E_CORE_ERROR);
if ($isFatalError) {
    // xdebug_get_function_stack doesn't include the frame that triggered the error!
    $errorFileLine = array(
        'file' => $error['file'],  // cannot access array value on possibly null variable $error
        'line' => $error['line'], // cannot access array value on possibly null variable $error
    );
}
```

psalm knows `$error` isn't `null`  when accessing `$error['type']`..  but doesn't seem to know inside the condition

**Comment by muglug** [comment]
Yeah, it doesn't support dependent types – it doesn't know that `$isFatalError` is only true when `$error !== null`, and probably won't ever.

**Comment by bkdotcom** [comment]
less verbose, but doesn't complain:

```php
$error = \error_get_last();
if ($error !== null && $error['type'] & (E_ERROR | E_PARSE | E_COMPILE_ERROR | E_CORE_ERROR);) {
    // xdebug_get_function_stack doesn't include the frame that triggered the error!
    $errorFileLine = array(
        'file' => $error['file'],  // all good
        'line' => $error['line'], // all good
    );
}
```
